backend/src/dfini_hack_team4.did

```diff
@@ -0,0 +1,59 @@
+type Amount = nat64;
+type AnswerId = nat64;
+type Solution = blob;
+type TaskId = nat64;
+type TaskPayload = blob;
+type Timestamp = nat64;
+
+type TaskType = variant {
+  translate_text;
+  edit_image;
+};
+
+type TaskStatus = variant {
+  open;
+  closed;
+};
+
+type Choice = variant {
+  yes;
+  no;
+};
+
+type Vote = record {
+  voter: principal;
+  choice: Choice;
+};
+
+type Answer = record {
+  submitter: principal;
+  submission_time: Timestamp;
+  solution: Solution;
+  votes: vec Vote;
+};
+
+type Task = record {
+  submitter: principal;
+  type: TaskType;
+  payload: TaskPayload;
+  deadline: Deadline,
+  reward: Reward,
+  answers: vec Answer;
+  status: TaskStatus,
+};
+
+type ShortTask = record {
+  id: TaskId;
+  submitter: principal;
+  type: TaskType;
+};
+
```

Comment:
do we also need a type or so for the ledger / balance?

Reply:
I believe this will mostly be needed internally, so I'd leave it out from the public interface of the canister. Even the debug API we discussed `get_ledger` was not included here because I don't want to have APIs that are supposed to be internal/for debugging purposes mixed with the public facing APIs.